centroid_decompostition.cpp

```diff
@@ -0,0 +1,92 @@
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+class Graph 
+{
+private:
+	int ver;
+	vector<int>* adj;
+	vector<int>* centroidTree;
+	int* sizes;
+	bool* deleted;
+public:
+	Graph(int num) : ver(num) {
+		adj = new vector<int>(ver + 1);
+		centroidTree = new vector<int>(ver + 1);
+		sizes = new int[ver + 1];
+		deleted = new bool[ver + 1];	
+	}
+	
+	~Graph();
+	
+	void addEdge(int x, int y, vector<int> arr[]);
+
+	void calcSizes(int cur, int par);
+
+	int findCentroid(int cur, int par, int vertices);
+
+	void decomposeTree(int cur, int total);
+};
+
+Graph::~Graph()
+{
+	delete []adj;
+	delete []sizes;
+	delete []deleted;
+}
+
+void Graph::addEdge(int x, int y, vector<int> arr[])
```

Comment:
Fix this function to:

```C++
void Graph::addEdge(int x, int y, std::vector<std::vector<int>>& graph) // Or whatever name. 
                                                                        // Make sure to pass by reference,
                                                                        // since you will modify the vector
{
    // If you want, you can do bounds checking:
    if (x >= graph.size() || x < 0)
    {
        throw std::out_of_range{"x coordinate out of bounds"}; // http://en.cppreference.com/w/cpp/error/out_of_range
    }
    graph[x].push_back(y);
```